### PR TITLE
Isomorphic Strings Algorithm

### PR DESCRIPTION
**Pull request by jsca-kwok**
This PR proposes to add a hash solution to check for isomorphic strings

### CODE REVIEW

File: data_structures/hash_table/isomorphic_strings.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+# Input: s = "paper", t = "title"
+# Output: true
+#
+# Constraints:
+# 1 <= s.length <= 5 * 104
+# t.length == s.length
+# s and t consist of any valid ascii character.
+
+# Approach 1: Hash Map
+# Time Complexity: O(N)
```
Comment (vbrazo) [inline data_structures/hash_table/isomorphic_strings.rb:26]: What's the space complexity of this algorithm?

Comment (jsca-kwok) [reply data_structures/hash_table/isomorphic_strings.rb:26]: > What's the space complexity of this algorithm?

Would it be O(N) because of the loop?

Comment (vbrazo) [reply data_structures/hash_table/isomorphic_strings.rb:26]: In this case, I believe you'd have a space complexity equal to N + M where N is the size of the `s` and M the size of the `r` array.

Comment (vbrazo) [reply data_structures/hash_table/isomorphic_strings.rb:26]: I think you're right. It's`O(N)` where N is the length of the `s` array.

**Review by vbrazo — APPROVED** [review]

**Pull request merged**